src/NuGet.Core/NuGet.Protocol/HttpSource/HttpSource.cs

```diff
@@ -83,6 +83,25 @@ public class HttpSource : IHttpSource
                 request.CacheKey,
                 request.CacheContext);
 
+            //////////////////////////////////////////////////////////
+            // Start - Chocolatey Specific Modification
+            //////////////////////////////////////////////////////////
+
+            // There are times when the URI that is being queried results in a CacheFile name that
+            // is greater that the allowed 260 characters for a file path.  While .NET "should" handle
+            // this, some methods in the BCL don't, and as a result, we need to create a hashed
+            // version of the file path, so that we can cache the file with that name, and therefore
+            // an exception will not be thrown.
+            if (cacheResult.NewFile.Length > 260 || cacheResult.CacheFile.Length > 260)
+            {
+                cacheResult.NewFile = GetHashedCacheFileName(cacheResult.NewFile);
+                cacheResult.CacheFile = GetHashedCacheFileName(cacheResult.CacheFile);
+            }
```

Comment:
Do we need to do both of them?
If one is above 260 characters, the other will also always be over 260 characters?

Would it perhaps be better to do

```suggestion
            if (cacheResult.NewFile.Length > 260)
            {
                cacheResult.NewFile = GetHashedCacheFileName(cacheResult.NewFile);
            }

            if (cacheResult.CacheFile.Length > 260)
            {
                cacheResult.CacheFile = GetHashedCacheFileName(cacheResult.CacheFile);
            }
```

Reply:
Makes sense to me.